test/asyncio_tests/test_asyncio_basic.py

```diff
@@ -130,3 +130,19 @@ class C(ABC):
         # MOTOR-104, TypeError: Can't instantiate abstract class C with abstract
         # methods collection, db, subcollection.
         C()
+
+    @asyncio_test
+    async def test_inheritance(self):
+        class CollectionSubclass(motor_asyncio.AsyncIOMotorCollection):
+            pass
+
+        class DatabaseSubclass(motor_asyncio.AsyncIOMotorDatabase):
+            def __getitem__(self, name):
+                return CollectionSubclass(self, name)
+
+        class ClientSubclass(motor_asyncio.AsyncIOMotorClient):
+            def __getitem__(self, name):
+                return DatabaseSubclass(self, name)
+
+        cx = ClientSubclass(test.env.uri, **self.get_client_kwargs())
+        self.assertIsNotNone(await cx.testdb.testcoll.insert_one({}))
```

Comment:
Neat test! One bug is that `cx.testdb.testcoll` uses `__getattr__ ` not `__getitem__`. Can we assert that the database and collection are actually DatabaseSubclass and CollectionSubclass?

Reply:
`__getattr__` for these classes actually calls `__getitem__` which is why I wrote it this way. But you are right, better to use the method directly to make the test less confusing. Updated and added type checks as well.